cocotb/triggers.py

```diff
@@ -526,27 +526,28 @@ class Combine(_AggregateWaitable):
     """
     Waits until all the passed triggers have fired.
 
-    Like most triggers, this simply returns itself.
+    Returns a list of results from each coroutine in the order they are provided.
     """
     @decorators.coroutine
     def _wait(self):
         waiters = []
         e = Event()
         triggers = list(self.triggers)
+        results = [None] * len(triggers)
 
         # start a parallel task for each trigger
-        for t in triggers:
+        for i, t in enumerate(triggers):
             # t=t is needed for the closure to bind correctly
-            def on_done(ret, t=t):
+            def on_done(ret, i=i, t=t):
                 triggers.remove(t)
                 if not triggers:
                     e.set()
-                ret.get()  # re-raise any exception
+                results[i] = ret.get()  # re-raise any exception
             waiters.append(cocotb.fork(_wait_callback(t, on_done)))
 
         # wait for the last waiter to complete
         yield e.wait()
-        raise ReturnValue(self)
+        raise ReturnValue(results)
```

Comment:
Ah, I see you're ahead of me here :)

I don't think we can make this change to Combine, it breaks usages like `First(a, Combine(b, c))`.

I'd opt for adding an extra kwarg to `Combine.__init__`, `return_results=True`

Reply:
I think, if this is a significant departure from the typical Trigger interface, I might revert to my previous double join method and leave you to decide if this is worth doing :P

Reply:
I think that's reasonable - it's clearly a nice thing to have, but probably does not belong in this PR.